x-pack/elastic-agent/pkg/composable/providers/host/host.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package host
+
+import (
+	"os"
+	"reflect"
+	"runtime"
+	"time"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+	"github.com/elastic/go-sysinfo"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+)
+
+// DefaultCheckTimeout is the default timeout used to check if any host information has changed.
+const DefaultCheckTimeout = 5 * time.Minute
```

Comment:
I wonder if we should make this configurable? It is more to have an example on how configs in a provider work.

Reply:
I made it configurable.